streamer/pulp/streamer/server.py

```diff
@@ -52,6 +52,36 @@ class DownloadListener(AggregatingEventListener):
     Nectar download listener.
     """
 
+    def __init__(self, streamer, request):
+        """
+        :param streamer: The streamer.
+        :type  streamer: Streamer
+        :param request: An HTTP request.
```

Comment:
The type is a _twisted_ Request which is always the original client request being handled by the streamer.